File: dto/dtos/State.cs
using dto.Paging;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Security.Cryptography.X509Certificates;

namespace dto.dtos
{
    public class StateRes
    {
        public int StateId { get; set; }
        public string? Slug { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
        public CountryRes? Country { get; set; }

        // Child tables
        public ICollection<CityRes>? Cities { get; set; }
        public IEnumerable<StateNameRes>? StateNames { get; set; }
    }
    public class StateReqEdit
    {
        [Required]
        public string? Slug { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
    }
    public class StateReqSearch : PagedReq
    {
        public int? CountryId { get; set; }
    }
}

File: admin/src/dtos/State.ts
import Common from "../utility/Common";
import { CityRes } from "./City";
import { CountryRes } from "./Country";
import { PagedReq } from "./PagedReq";
import { StateNameRes } from "./StateName";

export interface StateRes {
  stateId?: string;
  slug?: string;
  countryId?: string;
  country?: CountryRes;

  cities?: CityRes[];
  stateNames?: StateNameRes[];
}

export class StateReqEdit {
  slug?: string = "";
  countryId?: string = "";

  constructor(countryId?: string) {
    this.countryId = countryId;
  }
}

export class StateReqSearch extends PagedReq {
  countryId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {countryId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.countryId = countryId;
  }
}